include/nbl/asset/utils/IMeshPackerV2.h

```diff
@@ -539,34 +610,45 @@ class IMeshPackerV2 : public IMeshPacker<MeshBufferType,MDIStructType>, public I
-    template <typename MeshBufferIterator>
-    uint32_t calcMDIStructMaxCount(const MeshBufferIterator mbBegin, const MeshBufferIterator mbEnd);
-
     inline const PackerDataStore& getPackerDataStore() const { return m_packerDataStore; }
-    
-    inline const AllocationParams& getAllocParams() const { return m_allocParams; }
+
+    const core::GeneralpurposeAddressAllocator<uint32_t>& getMDIAllocator() const { return m_MDIDataAlctr; }
+    const core::GeneralpurposeAddressAllocator<uint32_t>& getIndexAllocator() const { return m_idxBuffAlctr; }
+    const core::GeneralpurposeAddressAllocator<uint32_t>& getVertexAllocator() const { return m_vtxBuffAlctr; }
 
 protected:
-	IMeshPackerV2(const AllocationParams& allocParams, uint16_t minTriangleCountPerMDIData, uint16_t maxTriangleCountPerMDIData)
-		: base_t(minTriangleCountPerMDIData, maxTriangleCountPerMDIData), m_allocParams(allocParams)
+	IMeshPackerV2(const AllocationParams& allocParams, const SupportedFormatsContainer& formats, uint16_t minTriangleCountPerMDIData, uint16_t maxTriangleCountPerMDIData)
+		: base_t(minTriangleCountPerMDIData, maxTriangleCountPerMDIData), 
+        IMeshPackerV2Base(formats)
 	{
         initializeCommonAllocators(allocParams);
     };
     template<class OtherBufferType, class OtherDescriptorSetType, class OtherMeshBufferType>
 	explicit IMeshPackerV2(const IMeshPackerV2<OtherBufferType,OtherDescriptorSetType,OtherMeshBufferType,MDIStructType>* otherMp)
 		: base_t(otherMp->getMinTriangleCountPerMDI(),otherMp->getMaxTriangleCountPerMDI()),
-        IMeshPackerV2Base(otherMp->getVirtualAttribConfig()), m_allocParams(otherMp->getAllocParams())
+        IMeshPackerV2Base(otherMp->getVirtualAttribConfig())
 	{
-        initializeCommonAllocators(m_allocParams);
+
+        const auto& mdiAlctr = otherMp->getMDIAllocator();
+        uint32_t alctrBuffSz = alctrTraits::get_total_size(mdiAlctr);
+        void* resSpcTmp = _NBL_ALIGNED_MALLOC(alctrTraits::reserved_size(alctrBuffSz, mdiAlctr), _NBL_SIMD_ALIGNMENT);
+        m_MDIDataAlctr = core::GeneralpurposeAddressAllocator<uint32_t>(alctrBuffSz, mdiAlctr, resSpcTmp);
+
+        const auto& idxAlctr = otherMp->getIndexAllocator();
+        alctrBuffSz = alctrTraits::get_total_size(idxAlctr);
+        resSpcTmp = _NBL_ALIGNED_MALLOC(alctrTraits::reserved_size(alctrBuffSz, idxAlctr), _NBL_SIMD_ALIGNMENT);
+        m_idxBuffAlctr = core::GeneralpurposeAddressAllocator<uint32_t>(alctrBuffSz, idxAlctr, resSpcTmp);
+
+        const auto& vtxAlctr = otherMp->getVertexAllocator();
+        alctrBuffSz = alctrTraits::get_total_size(vtxAlctr);
+        resSpcTmp = _NBL_ALIGNED_MALLOC(alctrTraits::reserved_size(alctrBuffSz, vtxAlctr), _NBL_SIMD_ALIGNMENT);
+        m_vtxBuffAlctr = core::GeneralpurposeAddressAllocator<uint32_t>(alctrBuffSz, vtxAlctr, resSpcTmp);
```

Comment:
I think this is why I suggested an overload of `initializeCommonAllocators` which takes const references to the allocators we're supposed to copy